### ISSUE TITLE
Failed to load plugin descriptor from file

### ISSUE THREAD

**Issue by mikewenzel** [Camelcade/Perl5-IDEA#2036]
When I try to install plugin from disk I get the error message `Failed to load plugin descriptor from file perl5-2019.1.zip`.

I cloned the repo to disk and did the gradle command listed. Output:

```
username@MacBook-Pro ~/GIT/intellij-perl/Perl5-IDEA (master) ▸ ./gradlew buildPlugin
Downloading https://services.gradle.org/distributions/gradle-4.10.2-all.zip
..................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
Unzipping /Users/username/.gradle/wrapper/dists/gradle-4.10.2-all/9fahxiiecdb76a5g3aw9oi8rv/gradle-4.10.2-all.zip to /Users/username/.gradle/wrapper/dists/gradle-4.10.2-all/9fahxiiecdb76a5g3aw9oi8rv
Set executable permissions for: /Users/username/.gradle/wrapper/dists/gradle-4.10.2-all/9fahxiiecdb76a5g3aw9oi8rv/gradle-4.10.2/bin/gradle

Welcome to Gradle 4.10.2!

Here are the highlights of this release:
 - Incremental Java compilation by default
 - Periodic Gradle caches cleanup
 - Gradle Kotlin DSL 1.0-RC6
 - Nested included builds
 - SNAPSHOT plugin versions in the `plugins {}` block

For more details see https://docs.gradle.org/4.10.2/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

> Task :perl5.core:generatePerl5Parser
BNF_RULE:deref_expr: 'expr ( <op> expr) +' expected for N-ary operator, treating as POSTFIX
BNF_RULE:redo_expr: unexpected cardinality OPTIONAL of expr, treating as ATOM
BNF_RULE:goto_expr: unexpected cardinality OPTIONAL of expr, treating as ATOM
BNF_RULE:next_expr: unexpected cardinality OPTIONAL of expr, treating as ATOM
BNF_RULE:last_expr: unexpected cardinality OPTIONAL of expr, treating as ATOM
Perl5.bnf parser generated to /Users/username/GIT/intellij-perl/Perl5-IDEA/core/gen

> Task :perl5.core:generatePerlLexer
Reading skeleton file "/Users/username/GIT/intellij-perl/Perl5-IDEA/grammar/Perl5.skeleton".
Reading "/Users/username/GIT/intellij-perl/Perl5-IDEA/core/grammar/Perl.flex"
Constructing NFA : 8,172 states in NFA
Converting NFA to DFA : 
...............................................................................................................................................................................XECUTING [1m 58s]......................................................................................................................6632 states before minimization, 4481 states in minimized DFA..............................................................................................Writing code to "/Users/username/GIT/intellij-perl/Perl5-IDEA/core/gen/com/perl5/lang/perl/lexer/PerlLexer.java".................................................
> Task :perl5.core:generatePodParser
Pod.bnf parser generated to /Users/username/GIT/intellij-perl/Perl5-IDEA/core/gen

> Task :perl5.core:generatePodLexer
Reading skeleton file "/Users/username/GIT/intellij-perl/Perl5-IDEA/grammar/Perl5.skeleton".
Reading "/Users/username/GIT/intellij-perl/Perl5-IDEA/core/grammar/Pod.flex"
Constructing NFA : 282 states in NFA
Converting NFA to DFA : 
..........................................................................................................................................................
163 states before minimization, 128 states in minimized DFA
Writing code to "/Users/username/GIT/intellij-perl/Perl5-IDEA/core/gen/com/perl5/lang/pod/lexer/PodLexerGenerated.java"

> Task :perl5.core:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :perl5.lang.embedded:generateEmbeddedPerlLexer
Reading skeleton file "/Users/username/GIT/intellij-perl/Perl5-IDEA/grammar/Perl5.skeleton".
Reading "/Users/username/GIT/intellij-perl/Perl5-IDEA/embedded/grammar/EmbeddedPerl.flex"
Constructing NFA : 64 states in NFA
Converting NFA to DFA : 
............................
36 states before minimization, 19 states in minimized DFA
Writing code to "/Users/username/GIT/intellij-perl/Perl5-IDEA/embedded/gen/com/perl5/lang/embedded/lexer/EmbeddedPerlLexer.java"

> Task :perl5.lang.mason.htmlmason:generateHTMLMasonLexer
Reading skeleton file "/Users/username/GIT/intellij-perl/Perl5-IDEA/grammar/Perl5.skeleton".
Reading "/Users/username/GIT/intellij-perl/Perl5-IDEA/htmlmason/grammar/HTMLMason.flex"
Constructing NFA : 828 states in NFA
Converting NFA to DFA : 
..........................................................................................................................................................................................................................................................................................................................................................................................................................................................................
532 states before minimization, 378 states in minimized DFA
Writing code to "/Users/username/GIT/intellij-perl/Perl5-IDEA/htmlmason/gen/com/perl5/lang/htmlmason/lexer/HTMLMasonLexer.java"

> Task :perl5.lang.mason.htmlmason:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/username/GIT/intellij-perl/Perl5-IDEA/htmlmason/src/com/perl5/lang/htmlmason/parser/psi/impl/HTMLMasonFileImpl.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :perl5.lang.mason.mason2:generateMason2Lexer
Reading skeleton file "/Users/username/GIT/intellij-perl/Perl5-IDEA/grammar/Perl5.skeleton".
Reading "/Users/username/GIT/intellij-perl/Perl5-IDEA/mason2/grammar/Mason2.flex"
Constructing NFA : 720 states in NFA
Converting NFA to DFA : 
........................................................................................................................................................................................................................................................................................................................................................................................................................
458 states before minimization, 296 states in minimized DFA
Writing code to "/Users/username/GIT/intellij-perl/Perl5-IDEA/mason2/gen/com/perl5/lang/mason2/lexer/Mason2TemplatingLexer.java"

> Task :perl5.lang.mason.mason2:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :perl5.lang.mojo:generateMojoliciousLexer
Reading skeleton file "/Users/username/GIT/intellij-perl/Perl5-IDEA/grammar/Perl5.skeleton".
Reading "/Users/username/GIT/intellij-perl/Perl5-IDEA/mojo/grammar/Mojolicious.flex"
Constructing NFA : 300 states in NFA
Converting NFA to DFA : 
..........................................................................................................................................................
176 states before minimization, 84 states in minimized DFA
Writing code to "/Users/username/GIT/intellij-perl/Perl5-IDEA/mojo/gen/com/perl5/lang/mojolicious/lexer/MojoliciousLexer.java"

> Task :perl5.lang.mojo:compileJava
Note: /Users/username/GIT/intellij-perl/Perl5-IDEA/mojo/src/com/perl5/lang/mojolicious/idea/editor/smartkeys/MojoliciousSmartKeysUtil.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :perl5.lang.tt2:generateTT2Parser
TemplateToolkit.bnf parser generated to /Users/username/GIT/intellij-perl/Perl5-IDEA/tt2/gen

> Task :perl5.lang.tt2:generateTT2Lexer
Reading skeleton file "/Users/username/GIT/intellij-perl/Perl5-IDEA/grammar/Perl5.skeleton".
Reading "/Users/username/GIT/intellij-perl/Perl5-IDEA/tt2/grammar/TemplateToolkit.flex"
Constructing NFA : 134 states in NFA
Converting NFA to DFA : 
.....................................................
67 states before minimization, 49 states in minimized DFA
Writing code to "/Users/username/GIT/intellij-perl/Perl5-IDEA/tt2/gen/com/perl5/lang/tt2/lexer/TemplateToolkitLexerGenerated.java"

> Task :perl5.lang.tt2:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :perl5.transport.docker:compileJava
Note: /Users/username/GIT/intellij-perl/Perl5-IDEA/docker/src/com/perl5/lang/perl/idea/sdk/host/docker/PerlDockerHandler.java uses or overrides a deprecated API
Note: Recompile with -Xlint:deprecation for details.

> Task :perl5.transport.wsl:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Download https://cache-redirector.jetbrains.com/jetbrains.bintray.com/intellij-jbr/jbrx-8u202-osx-x64-b1483.39.tar.gz

> Task :buildSearchableOptions
Starting searchable options index builder
Searchable options index builder completed

> Task :perl5.core:buildSearchableOptions
Starting searchable options index builder
Searchable options index builder completed

> Task :perl5.lang.embedded:buildSearchableOptions
Starting searchable options index builder
Searchable options index builder completed

> Task :perl5.lang.mason.framework:buildSearchableOptions
Starting searchable options index builder
Searchable options index builder completed

> Task :perl5.lang.mason.htmlmason:buildSearchableOptions
Starting searchable options index builder
Searchable options index builder completed

> Task :perl5.lang.mason.mason2:buildSearchableOptions
Starting searchable options index builder
Searchable options index builder completed

> Task :perl5.lang.mojo:buildSearchableOptions
Starting searchable options index builder
Searchable options index builder completed

> Task :perl5.lang.tt2:buildSearchableOptions
Starting searchable options index builder
Searchable options index builder completed

> Task :perl5.transport.docker:buildSearchableOptions
Starting searchable options index builder
Searchable options index builder completed

> Task :perl5.transport.wsl:buildSearchableOptions
Starting searchable options index builder
Searchable options index builder completed

BUILD SUCCESSFUL in 6m 23s
106 actionable tasks: 106 executed
username@MacBook-Pro ~/GIT/intellij-perl/Perl5-IDEA (master) ▸
```

Am I doing something wrong?

**Comment by hurricup** [comment]
This zip is empty and just a side-effect of project structure. Each module of this project is a plugin. Main part is in the `core` module. See `core/build/distributions/...`. 
Btw, just curious, why not install from the Jetbrains repo?

**Comment by mikewenzel** [comment]
> Btw, just curious, why not install from the Jetbrains repo?

Error in front of keyboard I guess. Currently using IntelliJ only as second IDE for coding. Never noticed I installed only minor updates and never changed to the current major version. In my Jetbrain repo-list I never got the new versions. Sorry for that issue.

**Comment by hurricup** [comment]
Do you have the last version of IDEA? I've seen such behavior on obsolete versions only.

**Comment by hurricup** [comment]
Also, please note, that every new version of perl plugin released for specific version of JetBrains products. E..g you can't install 2019.1 on IDEA 2018.1

**Comment by hurricup** [comment]
Btw, what is your main IDE and why (if it's not a secret). And what language you are coding on mainly?

**Comment by mikewenzel** [comment]
> Do you have the last version of IDEA? I've seen such behavior on obsolete versions only.
Now I do have the latest version.

> Also, please note, that every new version of perl plugin released for specific version of JetBrains products. E..g you can't install 2019.1 on IDEA 2018.1
Yeah I noticed this when I tried to install it from disk after knowing the right directory for the file.

> Btw, what is your main IDE and why (if it's not a secret). And what language you are coding on mainly?
Currently I mostly code in perl. It's kind of a comfort thing: I often use an depoyment plugin which deploys (on save) scripts to the server by using ssh. As far as I know there was a deployment plugin for IntelliJ in the past which is now part of the Ultimate version. I'm not sure if I really need the ultimate edition for coding perl. That's basically the only reason it's not my main currently.
